soccer/gameplay/tactics/advance_zone_midfielder.py

```diff
@@ -0,0 +1,174 @@
+        self.add_transition(AdvanceZoneMidfielder.State.passSet,
+                            AdvanceZoneMidfielder.State.hold, lambda: False,
+                            "When a kick play begins")
+        self.add_transition(AdvanceZoneMidfielder.State.hold,
+                            AdvanceZoneMidfielder.State.passSet, lambda: True,
+                            "When not in a kick play")
+
+    def on_enter_passSet(self):
+        print("Entering PassSet");
+
+    def execute_passSet(self):
+        #gets the best position to travel to for ball reception
+        best_Pos_From_Ball = self.passing_point
+        """if (self.dribble_state):
+            points = [robocup.Point(0, 0), robocup.Point(0, 0)]
+            self.optimal = 1
+            # sets either the left or the right side of the ball to the best ball position
+            # and then notes which one is the other
+            # left side of the ball
+            points[self.optimal] = best_Pos_From_Ball
+            self.other = 0
+        else :
+            points = [robocup.Point(0, 0), robocup.Point(0, 0)]
+            self.optimal = 1
+            # sets either the left or the right side of the ball to the best ball position
+            # and then notes which one is the other
+            # left side of the ball
+            points[self.optimal] = best_Pos_From_Ball
+            self.other = 0
+        """
+        points = [robocup.Point(0, 0), robocup.Point(0, 0)]
+        points[0] = best_Pos_From_Ball;
+
+        # sets the second point
+        pointer, value2 = evaluation.passing_positioning.eval_best_receive_point(
+                best_Pos_From_Ball,
+                main.our_robots(), AdvanceZoneMidfielder.FIELD_POS_WEIGHTS,
+                AdvanceZoneMidfielder.NELDER_MEAD_ARGS,
+                AdvanceZoneMidfielder.PASSING_WEIGHTS)
+        
+        points[1] = pointer
+        # moves the robots
+        for i in range(2):
+            if (self.moves[i] is None):
+                self.moves[i] = skills.move.Move(points[i])
+                self.add_subbehavior(
+                    self.moves[i], self.names[i], required=False, priority= self.priorities[i])
+            else:
+                self.moves[i].pos = points[i]
+
+    def on_enter_hold(self):
+        print("kicking")
+
+    def execute_hold(self):
+
+        y_temp_hold = 0.8 * self.passing_point.y
+
+        x_temp_hold = constants.Field.Width / 3
+
+        self.hold_point = [robocup.Point(x_temp_hold,y_temp_hold), robocup.Point(x_temp_hold,y_temp_hold)]
+
+        for i in range(2):
+            if (self.moves[i] is None):
+                self.moves[i] = skills.move.Move(self.hold_point[i])
+                self.add_subbehavior(self.moves[i],
+                                     self.names[i],
+                                     required=False,
+                                     priority=1)
+            else:
+                self.moves[i].pos = self.hold_point[i]
+
+
+
+    @property
```

Comment:
You don't actually need all this property stuff, if you want to simplify stuff, we can set kick directly. I think you should just be able to remove the property and setter for all these variables and everything should just work.

these are mainly useful only if we want to change how a variable is get/set without using functions.

Reply:
I need it to receive a already calculated best point from adaptive formation. It greatly increases the efficiency of the class. I also need to know whether we are kicking to stop the passSet otherwise sometimes the best location to pass is directly in front of the clear kick to goal.